column_transforms/datepart/datepart.sql

```diff
@@ -1,3 +1,9 @@
 SELECT *,
 {% for col in date_columns %}DATE_PART('{{date_part}}', {{col}}) as {{col}}_datepart_{{date_part}} {{ ", " if not loop.last else "" }}{% endfor %}
+from {{ source_table }}
+
+SELECT *,
+{%- for target_col, date_part in dates.items() %}
+    DATE({{target_col}}, '{{date_part}}') as {{target_col}}_date {{ ", " if not loop.last else "" }}
+{%- endfor %}
```

Comment:
wait why are there two select statements here? Or are you planning to delete the first in favor of the second?

Reply:
Actually I think that you may have dropped a date cast into this template?

Reply:
thanks for the find